src/backend/distributed/connection/remote_commands.c

```diff
@@ -866,6 +872,7 @@ WaitForAllConnections(List *connectionList, bool raiseInterrupts)
 
 				if (connectionIsReady)
 				{
+					connectionIndex = event->pos + pendingConnectionsStartIndex;
```

Comment:
I am finding this difficult to follow

Reply:
found it out., not so straightforward though.

Reply:
added a comment